go_fixes_bug_5452_xml_omitempty.go

```diff
@@ -24,72 +28,61 @@ func (cco *CreditCardOptions) MarshalXML(e *xml.Encoder, start xml.StartElement)
 			VerificationMerchantAccountId string `xml:"verification-merchant-account-id,omitempty"`
 			UpdateExistingToken           string `xml:"update-existing-token,omitempty"`
 		}
-		return e.EncodeElement(
-			excludeVerifyCard{
-				VerifyCard:                    cco.VerifyCard,
-				VenmoSDKSession:               cco.VenmoSDKSession,
-				MakeDefault:                   cco.MakeDefault,
-				FailOnDuplicatePaymentMethod:  cco.FailOnDuplicatePaymentMethod,
-				VerificationMerchantAccountId: cco.VerificationMerchantAccountId,
-				UpdateExistingToken:           cco.UpdateExistingToken,
-			},
-			start,
-		)
-	} else {
-		type includeVerifyCard struct {
-			VerifyCard                    *bool  `xml:"verify-card"`
-			VenmoSDKSession               string `xml:"venmo-sdk-session,omitempty"`
-			MakeDefault                   bool   `xml:"make-default,omitempty"`
-			FailOnDuplicatePaymentMethod  bool   `xml:"fail-on-duplicate-payment-method,omitempty"`
-			VerificationMerchantAccountId string `xml:"verification-merchant-account-id,omitempty"`
-			UpdateExistingToken           string `xml:"update-existing-token,omitempty"`
-		}
-		return e.EncodeElement(
-			includeVerifyCard{
-				VerifyCard:                    cco.VerifyCard,
-				VenmoSDKSession:               cco.VenmoSDKSession,
-				MakeDefault:                   cco.MakeDefault,
-				FailOnDuplicatePaymentMethod:  cco.FailOnDuplicatePaymentMethod,
-				VerificationMerchantAccountId: cco.VerificationMerchantAccountId,
-				UpdateExistingToken:           cco.UpdateExistingToken,
-			},
-			start,
-		)
+		return e.EncodeElement(excludeVerifyCard(*cco), start)
+	}
+	type includeVerifyCard struct {
+		VerifyCard                    *bool  `xml:"verify-card"`
+		VenmoSDKSession               string `xml:"venmo-sdk-session,omitempty"`
+		MakeDefault                   bool   `xml:"make-default,omitempty"`
+		FailOnDuplicatePaymentMethod  bool   `xml:"fail-on-duplicate-payment-method,omitempty"`
+		VerificationMerchantAccountId string `xml:"verification-merchant-account-id,omitempty"`
+		UpdateExistingToken           string `xml:"update-existing-token,omitempty"`
 	}
+	return e.EncodeElement(includeVerifyCard(*cco), start)
```

Comment:
Unfortunately the ability to convert from one type to another that has the same fields and differs only with it's tags was introduced after Go 1.7, so we can't use it in the SDK yet.

Reply:
It is fixed now. I was not aware that the casting was relatively new feature of the language.